salt/modules/opkg.py

```diff
@@ -524,26 +524,35 @@ def install(name=None,
                 else:
                     to_install.append(pkgname)
             else:
-                pkgstr = '{0}={1}'.format(pkgname, version_num)
                 cver = old.get(pkgname, '')
-                if reinstall and cver and salt.utils.versions.compare(
-                        ver1=version_num,
-                        oper='==',
-                        ver2=cver,
-                        cmp_func=version_cmp):
-                    to_reinstall.append(pkgstr)
-                elif not cver or salt.utils.versions.compare(
-                        ver1=version_num,
-                        oper='>=',
-                        ver2=cver,
-                        cmp_func=version_cmp):
-                    to_install.append(pkgstr)
-                else:
-                    if not kwargs.get('only_upgrade', False):
-                        to_downgrade.append(pkgstr)
-                    else:
-                        # This should cause the command to fail.
+                version_conditions = [x.strip() for x in version_num.split(',')]
+                for version_condition in version_conditions:
+                    (version_string, version_operator, operator_specified) = _get_version_info(version_condition)
+                    if operator_specified:
+                        # Version conditions are sent to the solver as install commands
+                        pkgstr = '{0}{1}{2}'.format(pkgname, version_operator, version_string)
                         to_install.append(pkgstr)
+                        cmd_prefix.append('--force-downgrade')
```

Comment:
i don't think this is needed. see lines 504-505. we seem to add that by default. On the other side we don't add it if user requested upgrade_only which is also fine.